website/blog/2023-12-08-silver.md

```diff
@@ -0,0 +1,585 @@
+  </tr>
+</tbody>
+</table>
+
+For full details:
+[https://github.com/scalameta/metals/milestone/62?closed=1](https://github.com/scalameta/metals/milestone/62?closed=1)
+
+Metals is a language server for Scala that works with VS Code, Vim, Emacs and
+Sublime Text. Metals is developed at the [Scala Center](https://scala.epfl.ch/)
+and [VirtusLab](https://virtuslab.com) with the help from contributors from the
+community.
+
+## TL;DR
+
+Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
+give Metals a try!
+
+- [Use presentation compiler for synthetic decorations](#use-presentation-compiler-for-synthetic-decorations)
+- [Base Metals view on indexing information](#base-metals-view-on-indexing-information)
+- [Go to definition for scaladoc](#go-to-definition-for-scaladoc)
+- [Custom project root setting](#custom-project-root-setting)
+- [Bsp errors in status](#bsp-errors-in-status)
+- [Add back unintentionally removed support for 2.11.x](#add-back-unintentionally-removed-support-for-211x)
+- [Changes for plugin authors](#changes-for-plugin-authors)
+
+## Use presentation compiler for synthetic decorations
+
+Up until this release Metals would use the semanticdb information for displaying
+synthetic decorations such as inferred types or implicit conversions. This was
+not ideal as it meant that the information was not always up to date and could
+be inconsistent with the current state of code.
+
+![synthetics](https://i.imgur.com/td11vRH.gif)
+
+To fix that [jkciesluk](https://github.com/jkciesluk) switched to using the
+presentation compiler for this information. This means that each time code
+changes the decorations will be refreshed. This also added a number of new
+decorations that were not previously available for Scala 3 and it will be on par
+with the Scala 2 ones.
+
+## Base Metals view on indexing information
+
+Metals tree view enables users to navigate their workspace in a more structured
+way by directly showing the build modules and their packages, classes, objects
+etc. It would also allow users to navigate their dependency jars. Previously,
+this was based on special Scala 2 signature information, which means it wouldn't
+work in Scala 3.
+
+From this release we changed it to use the same information that is used for
+indexing in case of jars and semanticdb for local workspace files. This means
+that Scala 3 is fully supported and this has been the last big feature missing
+for full parity between Scala 2 and 3.
+
+![tree-view](https://i.imgur.com/Vun4kJh.gif)
+
+All the existing features should now work for Scala 3 as well. This includes the
+reveal command which allows to quickly reveal the current file in the tree view
+including inside jars.
+
+![tree-view-reveal](https://i.imgur.com/xW0W3I3.gif) Additionally, we improved
+icons used by the tree view. They will now use the default VS Code icons for the
+most part, which should make them more consistent with the rest of the editor.
```

Comment:
So I totally missed this change and now realize this is now broken in nvim-metals. Technically this is a breaking change for any client implementing the tree view protocol, since we now change what the `icon` names are. So in my case, no icons are showing since I had a mapping from the old names to the icons that corresponded with them.  I thought we weren't doing patch releases when we break things for clients?

I see down below there is a section for plugin authors and we just link to the Scala file with the new names. It'd be even more helpful to have something that lists the old names to new names. That'd make it much easier. For example

```
object -> symbol-object
...
```

Then authors don't need to dig through [this](https://github.com/scalameta/metals/commit/054fb2aa60650dcff089714efd271fa5c15cf22a#diff-168043e522e5797860de334e85bcb3dbce72414f4fc906ddd4928023d35182d6) to see what changed. I'd also recommend making mention that these icons correspond to the VS Code icons [here in the docs](https://scalameta.org/metals/docs/integrations/tree-view-protocol#tree-view-node).

Reply:
> Technically this is a breaking change for any client implementing the tree view protocol, since we now change what the icon names are. So in my case, no icons are showing since I had a mapping from the old names to the icons that corresponded with them. I thought we weren't doing patch releases when we break things for clients?

Ach sorry, I didn't check and I assumed that only VS Code used icons.

It will be easiest to revert the changes to icons and translate them in VS Code. We wouldn't need to change anything.

Reply:
😢 always forgetting that nvim has nice things 😢

Reply:
I forgot I also added some more types and I think it all is more consistent than before. Maybe we'll just do 1.2.0, because of that? What do you think?

Reply:
> 😢 always forgetting that nvim has nice things 😢

I am so sorry, I just thought it's terminal oriented so it's harder to get graphical stuff right. That was also due to my laziness :(

Reply:
Changed to 1.2.0